density/db/db.py

```diff
@@ -188,8 +189,8 @@ def get_oauth_code_for_uni(cursor, uni):
         return result['code']
     else:
         # If the code DNE, create a new one and insert into the database.
-        new_code = ''.join(random.choice(
-            string.ascii_uppercase + string.digits) for x in xrange(32))
+        token_bytes = os.urandom(32) + uuid.uuid4().bytes
+        new_code = base64.urlsafe_b64encode(token_bytes)
         query = """INSERT INTO oauth_data (uni, code)
                    VALUES (%s, %s);"""
```

Comment:
Yeah, that would definitely be a good idea; I'm not very familiar with injection-safe SQL practices (to be honest I'm not super familiar with SQL in general), but I can look into it at one point.